lib/private/FilesMetadata/Listener/MetadataDelete.php

```
@@ -28,17 +28,24 @@ public function __construct(
 	}
 
 	public function handle(Event $event): void {
-		if (!($event instanceof CacheEntryRemovedEvent)) {
+		if (!($event instanceof CacheEntriesRemovedEvent)) {
 			return;
 		}
 
-		try {
-			$nodeId = $event->getFileId();
-			if ($nodeId > 0) {
-				$this->filesMetadataManager->deleteMetadata($nodeId);
+		$entries = $event->getCacheEntryRemovedEvents();
+		$fileIds = [];
+
+		foreach ($entries as $entry) {
+			try {
+				$nodeId = $entry->getFileId();
+				if ($nodeId > 0) {
```

How could that be false? When is the event emitted for negative or null fileid?
I do not think the other code paths check this, we should not silently ignore it here without good reason.